### PR TITLE
Routes

### PR DESCRIPTION
**Pull request by CWells1225**
Created the controllers and routes. Testing routes

### CODE REVIEW

File: controllers/user.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+    if (schema.error) {
+        return res.status(StatusCodes.BAD_REQUEST).json({
+            error: schema.error.details[0].message,
+        });
+    }
+
+    const { fName, lName, email, password } = req.body; 
+
+    const userAlreadyExists = await User.findOne({ email: req.body.email });
```
Comment (mnebot86) [inline controllers/user.js]: u already deconstructed email from req.body on line 13. So you can just do   `await User.findOne({ email });`

File: controllers/user.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
+import User from '../models/user.js'; 
+import { StatusCodes } from 'http-status-codes';
+
+export const register = async (req, res) => {
+    const schema = registerSchema(req.body)
```
Comment (mnebot86) [inline controllers/user.js]: Where is registerSchema coming from?

File: controllers/user.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            error: '"Email" already in use', 
+            field: 'email',
+        });
+    }
+
+    const user = await User.create ({
+        fName, 
+        lName, 
+        email, 
+        password,
```
Comment (mnebot86) [inline controllers/user.js]: Your saving the password directly to the database. You never want to do this. The password should be hash and salted before storing. Look into the bcrypt library.

File: controllers/user.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    });
+
+    const token = user.createJWT();
```
Comment (mnebot86) [inline controllers/user.js]: `createJWT` was never created on the User model so this wont work.

File: controllers/user.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+    const { fName, lName, email, password } = req.body; 
+
+    const user = await User.findOne({ email }).select('+password');
+
+    if (!user) {
+        return res.json({ error: 'Invalid Credentials'}); 
+    }
+
+    const isPasswordCoreect = await user.comparePassword(password);
```
Comment (mnebot86) [inline controllers/user.js]: `comparePassword` wasn't created on the User model, so this wont work.

File: controllers/user.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+    if (!isPasswordCoreect) {
+        return res.json({ error: 'Invalid Credentials'});
+    }
+
+    const token = user.createJWT();
```
Comment (mnebot86) [inline controllers/user.js]: createJWT was never created on the User model so this wont work.

File: controllers/user.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+    const isPasswordCoreect = await user.comparePassword(password);
+
+    if (!isPasswordCoreect) {
+        return res.json({ error: 'Invalid Credentials'});
+    }
+
+    const token = user.createJWT(); 
+
+    user.password=undefined;
```
Comment (mnebot86) [inline controllers/user.js]: `user.password = undefined;`

File: controllers/user.js [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+    const { id } = req.params; 
+
+    if (!id) {
+            return res.json({ error: 'Please provide user id'});
+    }
+
+    await User.findByIdAndDelete(id);
+
+    return res.status(StatusCodes.OK).json({ message: 'Deleted!'});
+};
```
Comment (mnebot86) [inline controllers/user.js:84]: add new line

File: routes/user.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
+import express from 'express';
+import { 
+        register,
+	    login,
```
Comment (mnebot86) [inline routes/user.js]: fix alignment

File: routes/user.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+	    deleteUser, 
+    } from '../controllers/user.js';
+
+    const router = express.Router();
+
+    router.route('/register').post(register);
+    router.route('/login').post(login);
+    router.route('/users').get(checkAuth, getUsers);
+    router.route('/users/:id');
+    router.get(checkAuth, getUser);
```
Comment (mnebot86) [inline routes/user.js]: `checkAuth` function doesn't exist

**Review by mnebot86 — CHANGES_REQUESTED** [review]

**Comment by CWells1225** [comment]
I've updated the user controllers and routes. I started with the sign up controller / route .

**Pull request merged**